Merge forward 3007.x into master

dwoz opened the pull request:
This includes 3006.15 and 3007.7

gayathri-krishnaswamy committed:
Updated Windows package manager

(cherry picked from commit a5108d40f0144754e23aeb2944d4c8e56048e588)

gayathri-krishnaswamy committed:
Update windows-package-manager.rst

Updated comments provided by Alyssa

(cherry picked from commit 890889e1179267c3056405a081a40f43cc9ca10e)

gayathri-krishnaswamy committed:
Update windows-package-manager.rst

Updated some grammatical errors.

(cherry picked from commit 866f9598b9e5f2cb3bde3d27a527308db35d4a5a)

twangboy committed:
Fix a few typos and grammatical errors

(cherry picked from commit 4a45b06c16040fa9a456cc056b7433cbc4003599)

nicholasmhughes committed:
fixes saltstack/salt#66453 sync_renderers fails when the custom renderer is specified via config

nicholasmhughes committed:
fix pre-commit failures

meaksh committed:
Fix _pygit2.GitError: error loading known_hosts: issue on gitfs

This commit ensures the right HOME value is set during Pygit2 remote
initialization, otherwise there are chances to get exceptions as
pygit2/libgit2 is relying on it.

meaksh committed:
Add changelog entry file

meaksh committed:
Add test_checkout_pygit2_with_home_env_unset unit test

barneysowood committed:
Fix logging of key changes to be consistent

Fixes logging of master aes and session keys changes to be correct and
consistent. Also simplifies the code and reduces duplication.

DdangJin committed:
Ensure "test" returns only boolean values

DdangJin committed:
Add tests for LazyLoader options and ensure test value is boolean

# Conflicts:
#	tests/pytests/unit/loader/test_lazy.py

twangboy committed:
Remove duplicate test

DdangJin committed:
Add changelog

Commit:
fix

xsmile committed:
cmdmod: unset shell/python_shell variables by default

xsmile committed:
cmdmod: change argument processing/quoting

xsmile committed:
cmdmod: always remove the temporary script

xsmile committed:
cmdmod: set a default shell when running known script types

xsmile committed:
cmdmod: do not report success when a powershell command fails

xsmile committed:
cmdmod: update tests

xsmile committed:
cmdmod: early preparation of env dict

xsmile committed:
cmdmod: unify result processing for timed_subprocess and win_runas

xsmile committed:
win_runas: support more cmdmod parameters, fixups

xsmile committed:
win_runas: update tests

twangboy committed:
Update ansible to update ansible-core

twangboy committed:
Update zipp to 3.23.0

xsmile committed:
cmdmod: formatting and tests

xsmile committed:
cmdmod: changelog

xsmile committed:
add changelog

xsmile committed:
add docs

twangboy committed:
Fix some typos

twangboy committed:
Fix ansible tests

xsmile committed:
cmdmod: fix other modules

twangboy committed:
Don't write user registry with win_lgpo_reg

dwoz committed:
bad key logs useful message instead of traceback

When the minion's key is overwritten with bad data, log a useful message
instead of traceback. Handle the error in a consistant way accross
salt.minion, salt.channel.client, and salt.crypt.

Commit:
add changelog and unit test. Use context manager for DefaultSelector

twangboy committed:
Add force option to win_pkg

barneysowood committed:
Allow events to be processed in minion shutdown

Allows event to be handled after the salt-minion service has received
a SIGTERM. Previously once the signal handler was entered, the ioloop
would no longer run. If there are events on the minion event bus that
needs processing, they would not be handled.

Moves the MinionManager stop() functionality to an async function and
allows the ioloop to run and clear any waiting events and returns to
masters.

barneysowood committed:
Add test for MinionManager stop/stop_async

Adds test for calling MinionManager stop() to test new functionality to
allow events to be processed as minion is stopping.

Sets up a MinionManager instance with a running event bus, then calls
the stop function and immeadiately sends a test message on the event bus
and reads it back to check that works once the stop() function has been
called.

Then checks that the usual functions to destroy the minion etc have
also been called.

barneysowood committed:
Add changelog

barneysowood committed:
Updated docstrings to reflect the correct action

twangboy committed:
Fix state tests and a few bugs

barneysowood committed:
Modify systemd_service to use no_block for minion

Modifies systemd_service.{restart,stop} to default to using
no_block=True when the service being stopped or restarted is the
salt-minion.

If you don't pass no_block, the minion blocks waiting for systemd to
restart the service, while systemd is waiting for the minion to exit.
Eventually, after systemd hits its timeout it will kill the salt minion
processes.

Behaviour for other services should remain the same and the functions
will still honour the value of no_block if passed as an argument.

barneysowood committed:
Add changelog

barneysowood committed:
Ignore pylint warning on self.io_loop.add_callback

Ignores warning from pylint about self.io_loop.add_callback() not being
callable - it clearly is as stop_async gets called.

barneysowood committed:
Set up short sock_dir path in test

Sets up the short sock_dir path in the test. Previously setting it in
conftest.py was breaking another test because I'd done it as a Path,
not a string, but I want to avoid changin behaviour of other tests, so
setting it locally to the test.

twangboy committed:
Fixed some docs issues

barneysowood committed:
Update minion.restart to use service.restart

Updates minion.restart to use service.restart if system uses systemd or
is a Windows system.

Existing methods don't work if salt-minion is managed by the Windows
service manager or systemd. This adds functionality to check for those
and use service.restart if so.

barneysowood committed:
Add options to enable use of systemd/win_service

Adds options to enable use of systemd or windows service to restart the
salt minion. Both options default to True as we should prefer to use
the service managers, but give the option to disable where users are
doing something unusual like running another service supervisor.

dwoz committed:
decrease number of minion key checks

dwoz committed:
Memoize on all arguments

dwoz committed:
Ensure lock is released on request timeout

Ensure that message client lock is released by checking for the response
without blocking until the future provided to send_recv timesout.

vvrein committed:
Allow to pass context to pyobjects renderer

barneysowood committed:
Skip systemd test on Amazon Linux 2

Skips the test_operation_no_block_default test on Amazon Linux 2 as the
CI conatiner for that distro doesn't have a complete systemd setup.

barneysowood committed:
Add tests for minion.restart and helper functions

Adds unit tests for the minion.restart function to test existing and new
functionality. Also adds tests for helper functions.

barneysowood committed:
Add support for scheduled retry to start minion

Adds support for creating a scheduled job to retry starting the minion
process if it has died and not restarted correctly. This will work on
Windows and systemd based systems.

Commit:
modules/aptpkg.py: correct handling of foreign-only-arch packages (fixes #66940)

dwoz committed:
Use key modified time in singleton identifier

dwoz committed:
Bump relenv to 0.20.4

dwoz committed:
Revert "Ipc leak fix"

This reverts commit 4c0e5529f5d542570d87942902afa4ad1fe91b7c.

dwoz committed:
Add changelog for #68151

twangboy committed:
Enable signing macos packages

dwoz committed:
Merge branch '3006.x' into merge/3007.x/3006.x-25-08-04

dwoz committed:
Fix merge warts

dwoz committed:
Fix cmd test

dwoz committed:
Test fix

twangboy committed:
Enable signing macos packages

twangboy committed:
Fix sign logic... maybe

twangboy committed:
Merge pull request #68237 from twangboy/sign_macos_packages

Sign macos packages

twangboy committed:
Fix prepare signing

twangboy committed:
Merge pull request #68238 from twangboy/sign_macos_packages

Fix prepare signing

twangboy committed:
Use code suggested by github for setting up the keychain

twangboy committed:
Merge pull request #68239 from twangboy/sign_macos_packages

Use code suggested by github for setting up the keychain

twangboy committed:
Only delete the keychain if it was set up

twangboy committed:
Only delete the keychain if it was set up

twangboy committed:
Merge pull request #68240 from twangboy/sign_macos_packages

Only delete the keychain if it was set up

twangboy committed:
Merge branch '3006.x' into sign_macos_packages

twangboy committed:
Merge pull request #68241 from twangboy/sign_macos_packages

Only delete the keychain if it was set up

dwoz committed:
Remove crutfy workflow input

twangboy committed:
Revert "Only delete the keychain if it was set up"

This reverts commit 044f133db650489071a6b36fd15d2709b3263339.

twangboy committed:
Revert "Use code suggested by github for setting up the keychain"

This reverts commit ca4539b8cf8565219474690dd191421bf8a2ef4d.

twangboy committed:
Revert "Fix prepare signing"

This reverts commit 830330c68e95849f9fb7fc1481d3ac79b328137f.

twangboy committed:
Revert "Fix sign logic... maybe"

This reverts commit f31d7fe6df3920ff5c466801a5043e2f64b03077.

twangboy committed:
Revert "Enable signing macos packages"

This reverts commit 4342889550a3c42095c4d1c7fbd84f94fb065fbf.

twangboy committed:
Merge pull request #68244 from twangboy/sign_macos_packages

Sign macos packages

twangboy committed:
Fix macOS signing

twangboy committed:
Merge pull request #68245 from twangboy/sign_macos_packages

Fix macOS signing

Faelysse committed:
Fix max_depth not respected in file.directory state

Fixes: #55306

dwoz committed:
Allow test steps to be cancelled faster

dwoz committed:
No longer start daemons at module level

Module level fixtures do not yield with a started daemon. Keep the
module level fixtures around but start the daemons at the begining of
each test and stop them when the test is done. Doing this means each
test starts fresh and there is less likelyhood of affecting other tests
if something goes wrong.

dwoz committed:
Standardize auth singleton key

dwoz committed:
Re-enable ensure connections fixture

dwoz committed:
Fix dockermod tests

The default python interpereter in our containers has changed since
debian 13 has been released. Work around the shortcommings of our
containers and dockermod.

dwoz committed:
Merge remote-tracking branch 'saltstack/3006.x' into merge/3007.x/3006.x-25-08-04-mess

dwoz committed:
Merge pull request #68234 from dwoz/merge/3007.x/3006.x-25-08-04

Merge forward 3006.x into 3007.x

dwoz committed:
More responsive timeout and lock release

Fixing lock condition test in test. Make sure the lock release happens
as soon as the timeout happens rather than after a potential wait time
by waiting on the timeout future instead of a fixed amount of time.

dwoz committed:
Pull in scenario test improvments

dwoz committed:
Mark tests that require root

dwoz committed:
Clean up rogue print statements in tests

dwoz committed:
Yield to loop to allow server startup

dwoz committed:
Fix scenarios merge wart

twangboy committed:
Update CLI examples to conform with bash

twangboy committed:
Added changelog

twangboy committed:
Fix some docs errors

barneysowood committed:
Add unit tests for schedule_retry functionality

Adds unit tests for the schedule_retry functionality in the
minion.restart function.

Tests the helper functions that do the scheduling for windows and
systemd, tests the dispatcher function and also testsing the logic in
minion.restart itself.

barneysowood committed:
Refactor out duplicated code in minion.restart

Refactors minion.restart to move the duplicated code for systemd and
windows into a helper function

barneysowood committed:
Add changelogs for minion.restart changes

dwoz committed:
Use result_future to ensure we wait

Make sure the send_recv coroutine has finished by waiting on a result
future instead of the timeout future.

dwoz committed:
Merge 3006.x into 3007.x

dwoz committed:
Fix merge wart

dwoz committed:
Fix merge wart

dwoz committed:
Make sure pusher gets closed

dwoz committed:
Merge pull request #68278 from dwoz/merges/3007.x/3006.x-25-08-21

Merge 3006.x into 3007.x

dwoz committed:
Fix lint requirements for 3.11

lkubb committed:
Add test for issue #68281

lkubb committed:
Fix LazyLoader module handling

Commit e57901290ec46dfe600fd2e7cd4aa3983875f8cd
migrated away from the deprecated `load_module` to `exec_module`.
The latter does not reload modules like the deprecated function.

Since the loader assumes modules in `sys.modules` are never replaced, but
reloaded, ensure that's the case. There are likely better solutions in
theory, but this commit quickly reverts to the legacy behavior the
loader is used to.

If we replace modules instead, a mismatch between the cached functions
in a LazyLoader instance and the ones found on modules in `sys.modules`
can develop when we instantiate nested loaders of the same type and
load the same module with both.

dwoz committed:
Clean up tranport closing

dwoz committed:
Add changelog for #67076

dwoz committed:
Merge pull request #68279 from dwoz/issue/3006.x/62798

More cleanly close network connections

dwoz committed:
Merge pull request #68283 from dwoz/py311lintfix

Fix lint requirements for 3.11

dwoz committed:
Bump relenv version to 0.20.5

dwoz committed:
Add changelog for #68291

saltbot-open committed:
Release v3006.15

dwoz committed:
Merge branch '3006.x' into merge/3007.x/3006.x-25-08-27

dwoz committed:
Fix merge wart

dwoz committed:
Test fix

dwoz committed:
Merge pull request #68292 from dwoz/merge/3007.x/3006.x-25-08-27

Merge forward 3006.x into 3007.x

dwoz committed:
Merge branch '3006.x' into merge/3007.x/3006.x-25-08-28

dwoz committed:
Merge pull request #68295 from dwoz/merge/3007.x/3006.x-25-08-28

Merge forward form 3006.x to 3007.x

dwoz committed:
Fix merge wart

saltbot-open committed:
Release v3007.7

dwoz committed:
Merge branch '3007.x' into merge/master/3007.x-25-08-28

twangboy commented on doc/topics/windows/windows-package-manager.rst:
```diff
@@ -439,11 +439,11 @@ winrepo_dir_ng
 
 The location in the ``file_roots`` where the winrepo files are kept. The default
-is ``C:\ProgramData\Salt Project\Salt\srv\salt\win\repo-ng``.
+is ``C:\salt\srv\salt\win\repo-ng``.
```
It should be the ProgramData path

twangboy commented on doc/topics/windows/windows-package-manager.rst:
```diff
@@ -1333,13 +1333,16 @@ cache_dir (bool)
 This setting requires the software to be stored on the ``file_roots`` and only
+<<<<<<< HEAD
 applies to URLs that begin with ``salt://``\. If set to ``True``\, then the
+=======
```
Remove this conflict thing

dwoz committed:
Fix merge warts

dwoz committed:
Merge pull request #68284 from lkubb/fix/loader-sys-modules-prereq

Fix loader handling of sys.modules

barneysowood committed:
Add tests of log messages for master key rotation

Adds testing of debug log messasges for master key and session key
roatation. Added to existing test_auth_aes_key_rotation and
test_sauth_aes_key_rotation as otherwise I would have to duplicate most
of what they cover.

barneysowood committed:
Add changelog

dwoz committed:
The _get_opts helper method was removed.

dwoz committed:
test fixup

dwoz committed:
Fix merge wart

twangboy committed:
Update bootstrap script to version 2025.09.03

dwoz commented:
> @dwoz The current cmdmod behavior is that the `shell` and `python_shell` parameters are unset by default to prevent shell features from being enabled with `subprocess.Popen()`. They need to be explicitly enabled if required. `shell` will default to `/bin/bash` (the shell grain IIRC) if `python_shell` is set.
> 
> See the changes at https://github.com/saltstack/salt/pull/68156/files#diff-84a74747ac04938da8593d3d6092a2ae08afc91aef4532f9210932a6a00c7c95
> 
> test_avoid_injecting_shell_code_as_root https://github.com/saltstack/salt/blob/3006.x/tests/integration/modules/test_cmdmod.py#L438
> 
> test_hide_output (`dir` is a shell built-in on Windows while `ls` is a binary on Linux) https://github.com/saltstack/salt/blob/3006.x/tests/integration/modules/test_cmdmod.py#L576

The test failures indicate there has been a change in functionality since in the merge forward. Can you please take a look?

dwoz committed:
Refactor send recv to avoid race conditions and limit memory usage

dwoz commented:
> @dwoz The current cmdmod behavior is that the `shell` and `python_shell` parameters are unset by default to prevent shell features from being enabled with `subprocess.Popen()`. They need to be explicitly enabled if required. `shell` will default to `/bin/bash` (the shell grain IIRC) if `python_shell` is set.
> 
> See the changes at https://github.com/saltstack/salt/pull/68156/files#diff-84a74747ac04938da8593d3d6092a2ae08afc91aef4532f9210932a6a00c7c95
> 
> test_avoid_injecting_shell_code_as_root https://github.com/saltstack/salt/blob/3006.x/tests/integration/modules/test_cmdmod.py#L438
> 
> test_hide_output (`dir` is a shell built-in on Windows while `ls` is a binary on Linux) https://github.com/saltstack/salt/blob/3006.x/tests/integration/modules/test_cmdmod.py#L576

The test clearly show a change in functionality of `cmdmod.run_all` can you please take a look?

dwoz committed:
Use a tornado queue to reduce wait times in send recieve method

dwoz committed:
Hold on to send/recv socket reference.

Grab a reference to the send/recv socket so we don't loose it when the
close method is called.

dwoz committed:
Handle errors when loop closes before socket.

twangboy committed:
Use the name instead of the description to identify interfaces

twangboy committed:
Add tests, fix pre-commit failures

twangboy committed:
Add changelog

dwoz committed:
Add changelog for 68297.

dwoz committed:
Update relenv to 0.20.6

dwoz committed:
Fix polling timeout values

dwoz committed:
Fix pre-commit isort and black

dwoz committed:
Fix polling timeouts

twangboy committed:
Fix tests on Windows

twangboy committed:
Fixed failing tests on Linux, fix pre-commit

dwoz committed:
Fix typo

dwoz committed:
More tests for send_recv method.

twangboy committed:
Fix failing windows tests

twangboy committed:
Merge branch 'merge/master/3007.x-25-08-28' of https://github.com/dwoz/salt into dwoz-merge/master/3007.x-25-08-28

twangboy committed:
Use htop instead of algol68g

twangboy committed:
Remove ensurepip from packages

dwoz committed:
Clean up tests resources

dwoz committed:
Add test for receive socket close

twangboy committed:
Fix unit tests

twangboy committed:
Fix pre-commit, fix unit tests

dwoz committed:
Merge branch '3006.x' into merge/3006.x/3007.x-25-09-16

dwoz committed:
Fix tests and pre-commit

dwoz committed:
Port fix for #68297 to 3007.x

twangboy committed:
Fix test_win tests

dwoz committed:
Merge pull request #68334 from dwoz/merge/3006.x/3007.x-25-09-16

Merge forward 3006.x into 3007.x

saltbot-open committed:
Release v3006.16

saltbot-open committed:
Release v3007.8

dwoz committed:
Merge branch '3006.x' into merge/3007.x/3006.x-25-09-18

dwoz committed:
Merge pull request #68340 from dwoz/merge/3007.x/3006.x-25-09-18

Merge forward 3006.x into 3007.x

twangboy committed:
Fix typo

dwoz committed:
Merge remote-tracking branch 'saltstack/3007.x' into merge/master/3007.x-25-08-28

dwoz committed:
Check git test fix

dwoz committed:
Fix git config test on Windows

twangboy committed:
Fix pip-compile errors on pre-commit

twangboy committed:
Fix test_run on Windows

dwoz committed:
Merge branch '3006.x' into merge/3007.x/3006.x-15-09-22

dwoz committed:
Fix pre-commit

dwoz committed:
Merge pull request #68347 from dwoz/merge/3007.x/3006.x-25-09-22

Merge forward from 3006.x into 3007.x

dwoz committed:
Merge branch '3007.x' into merge/master/3007.x-25-08-28

dwoz committed:
Merge remote-tracking branch 'origin/merge/master/3007.x-25-08-28' into merge/master/3007.x-25-08-28

twangboy committed:
Fix pre-commit on 3007.x

dwoz committed:
Skip test due to dependency version upgrade

dwoz committed:
Merge remote-tracking branch 'saltstack/3007.x' into merge/master/3007.x-25-08-28

dwoz committed:
Pre-commit fix

dwoz committed:
Revert part of test fix

dwoz committed:
Fix regex argument escape

dwoz committed:
Fix version typo

dwoz committed:
more pre-commit fixes

dwoz committed:
Revert change to tests

dwoz committed:
Fix windows cmdmod.test_run on cicd
